sdk/core/azure-core/src/main/java/com/azure/core/util/paging/PageRetriever.java

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util.paging;
+
+import reactor.core.publisher.Flux;
+
+/**
+ * A type representing the contract to retrieve one or more pages.
+ *
+ * @param <C> the continuation token type
+ * @param <P> the page elements type
+ */
+@FunctionalInterface
+public interface PageRetriever<C, P> {
+    /**
+     * Retrieve a set of one or more pages starting from the page identified by
+     * the given continuation token.
+     *
+     * @param continuationToken the token identifying the page set, a {@code null}
+     *                          value indicate that retrieve pages from the beginning
+     * @param pageSize the preferred number of items per page, a {@code null} value
```

Comment:
Should we add some docs about what the behaviour will be if the pageSize is `< 1`?

Reply:
As Srikant suggested, we now use `null` instead of -1, so no more -1.